pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -382,6 +545,81 @@ func (n *NetworkPolicyController) processClusterNetworkPolicy(cnp *crdv1alpha1.C
 	return internalNetworkPolicy
 }
 
+// validateAppliedToWithSA validates if Spec.AppliedTo and Rule.AppliedTo are both
+// nil after trans ServiceAccounts to PodSelector+NamespaceSelector
+func (n *NetworkPolicyController) validateAppliedToWithSA(acnp *crdv1alpha1.ClusterNetworkPolicy) bool {
+	peerNotEmpty := func(peers []crdv1alpha1.NetworkPolicyPeer) bool {
+		for _, peer := range peers {
+			if len(peer.ServiceAccounts) == 0 {
+				return true
+			}
+			if len(n.transServiceAccounts(peer.ServiceAccounts)) > 0 {
+				return true
+			}
+		}
+		return false
+	}
+	if peerNotEmpty(acnp.Spec.AppliedTo) {
+		return true
+	}
+
+	allRulesHaveAT := func(rules []crdv1alpha1.Rule) bool {
+		for _, rule := range rules {
+			if !peerNotEmpty(rule.AppliedTo) {
+				return false
+			}
+		}
+		return true
+	}
+	return allRulesHaveAT(acnp.Spec.Ingress) && allRulesHaveAT(acnp.Spec.Egress)
+}
+
+// transServiceAccounts translates a list of ServiceAccount to a list of
+// NetworkPolicyPeer with PodSelector+NamespaceSelector.
+func (n *NetworkPolicyController) transServiceAccounts(serviceAccounts []crdv1alpha1.ServiceAccount) (newPeers []crdv1alpha1.NetworkPolicyPeer) {
```

Comment:
maybe just call it `translateServiceAccounts`...

Reply:
Changed.